Content.Client/Doors/DoorSystem.cs

```diff
@@ -1,28 +1,132 @@
+    [Dependency] private readonly IResourceCache _resourceCache = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<DoorComponent, AppearanceChangeEvent>(OnAppearanceChange);
+    }
+
+    protected override void OnComponentInit(EntityUid uid, DoorComponent comp, ComponentInit args)
+    {
+        comp.OpenSpriteStates = new(new (DoorVisualLayers, string)[1]{(DoorVisualLayers.Base, comp.OpenSpriteState)});
+        comp.ClosedSpriteStates = new(new (DoorVisualLayers, string)[1]{(DoorVisualLayers.Base, comp.ClosedSpriteState)});
+
+        comp.OpeningAnimation = new Animation()
+        {
+            Length = TimeSpan.FromSeconds(comp.OpeningAnimationTime),
+            AnimationTracks =
+            {
+                new AnimationTrackSpriteFlick()
+                {
+                    LayerKey = DoorVisualLayers.Base,
+                    KeyFrames = { new AnimationTrackSpriteFlick.KeyFrame(comp.OpeningSpriteState, 0f) }
+                }
+            },
+        };
+
+        comp.ClosingAnimation = new Animation()
+        {
+            Length = TimeSpan.FromSeconds(comp.ClosingAnimationTime),
+            AnimationTracks =
+            {
+                new AnimationTrackSpriteFlick()
+                {
+                    LayerKey = DoorVisualLayers.Base,
+                    KeyFrames = { new AnimationTrackSpriteFlick.KeyFrame(comp.ClosingSpriteState, 0f) }
+                }
+            },
+        };
+
+        comp.EmaggingAnimation = new Animation ()
+        {
+            Length = TimeSpan.FromSeconds(comp.EmaggingAnimationTime),
+            AnimationTracks =
+            {
+                new AnimationTrackSpriteFlick()
+                {
+                    LayerKey = DoorVisualLayers.BaseUnlit,
+                    KeyFrames = { new AnimationTrackSpriteFlick.KeyFrame(comp.EmaggingSpriteState, 0f) }
+                }
+            },
+        };
+    }
+
+    private void OnAppearanceChange(EntityUid uid, DoorComponent comp, ref AppearanceChangeEvent args)
     {
-        if (!Resolve(uid, ref door))
+        if (args.Sprite == null || !_gameTiming.IsFirstTimePredicted)
             return;
 
-        base.UpdateAppearance(uid, door);
+        if(!AppearanceSystem.TryGetData<DoorState>(uid, DoorVisuals.State, out var state, args.Component))
+            state = DoorState.Closed;
+
+        if (AppearanceSystem.TryGetData<string>(uid, DoorVisuals.BaseRSI, out var baseRsi, args.Component))
+        {
+            if (!_resourceCache.TryGetResource<RSIResource>(SharedSpriteComponent.TextureRoot / baseRsi, out var res))
+            {
+                Logger.Error("Unable to load RSI '{0}'. Trace:\n{1}", baseRsi, Environment.StackTrace);
+            }
+            foreach (ISpriteLayer layer in args.Sprite.AllLayers)
+            {
+                layer.Rsi = res?.RSI;
+            }
+        }
```

Comment:
i dont think this is ever used?

Reply:
Apparently it's used by the airlock painter.

Reply:
<https://github.com/space-wizards/space-station-14/blob/97a8b64c1d53077bdf62cfc5ce1a069381adb5da/Content.Server/AirlockPainter/AirlockPainterSystem.cs#L48>

Reply:
huh